Change pattern management title and text to avoid specifying pattern source incorrectly.

vcanales opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
Re. https://github.com/WordPress/gutenberg/issues/52498

Change wording for patterns in the editor to avoid mislabeling plugin patterns as theme patterns.

## Screenshots or screencast <!-- if applicable -->
<img width="348" alt="image" src="https://github.com/WordPress/gutenberg/assets/1157901/a24be425-bc54-4135-a16b-fb9f815862cb">

Commit:
Change "theme patterns" to "included patterns"

jameskoster commented:
Is "included" too ambiguous? 🤔 Maybe:

* Bundled patterns
* System patterns

It might be simpler to have no heading... (https://github.com/WordPress/gutenberg/pull/52570).

vcanales commented:
> It might be simpler to have no heading... (#52570).

Not sure about putting uneditable patterns with different sources under "My patterns"; it feels a bit confusing.

> Is "included" too ambiguous? 🤔 Maybe:
> 
> * Bundled patterns
> * System patterns

I like all of these! Happy to wait for @WordPress/gutenberg-design to chime in.

carolinan commented:
"Additional patterns"?

ndiego commented:
Can we just have the title "Patterns" or "Additional Patterns". My preference is just "Patterns". This does not lock us into any terminology, and we can iterate later. I think this needs to have at least some heading for 6.3. 

cc @richtabor @annezazu @glendaviesnz @carolinan thoughts?

vcanales commented:
> Can we just have the title "Patterns" or "Additional Patterns". My preference is just "Patterns". This does not lock us into any terminology, and we can iterate later. I think this needs to have at least some heading for 6.3.

Updated:
<img width="348" alt="image" src="https://github.com/WordPress/gutenberg/assets/1157901/72cd5768-588e-46cb-bf23-60ab72b56aaf">

I also changed the Read-only Pattern tooltip text to explain the situation a bit more; "Patterns bundled with themes or plugins cannot be edited." This is just a suggestion; happy to modify it as needed.

<img width="354" alt="image" src="https://github.com/WordPress/gutenberg/assets/1157901/cd50337f-feac-4813-9d76-ecb6955014cc">

annezazu commented:
I think just "Patterns" is too vague considering the entire section of the Site Editor is for Patterns. I think we need to go with some sort of identifier. Perhaps "Additional" if we want concise.

Commit:
Update theme & plugin patterns tooltip text

richtabor commented on packages/edit-site/src/components/sidebar-navigation-screen-patterns/index.js:
```diff
@@ -62,7 +62,7 @@ function ThemePatternsGroup( { categories, currentCategory, currentType } ) {
 	return (
 		<>
 			<div className="edit-site-sidebar-navigation-screen-patterns__group-header">
```
This was removed in https://github.com/WordPress/gutenberg/pull/52570, so no need to worry about the title here.

vcanales commented:
Seems like https://github.com/WordPress/gutenberg/pull/52570#event-9817707955 was already merged, removing the header. Still, I've updated it once more just in case.

<img width="351" alt="image" src="https://github.com/WordPress/gutenberg/assets/1157901/b308d83d-6ccf-4525-b1b6-9ac2fcee84c1">

richtabor commented on packages/edit-site/src/components/sidebar-navigation-screen-patterns/index.js:
```diff
@@ -72,7 +75,7 @@ function ThemePatternsGroup( { categories, currentCategory, currentType } ) {
 									position="top center"
 									text={ __(
-										'Theme patterns cannot be edited.'
+										'Patterns bundled with themes or plugins cannot be edited.'
```
We should just add the plugin reference like this: 

"Theme and plugin patterns cannot be edited."

richtabor replied:
That, or "Third-party patterns cannot be edited"

richtabor replied:
cc @annezazu thoughts?

richtabor replied:
@annezazu thoughts?

annezazu replied:
Agreed. That feels much cleaner and simpler.

vcanales commented:
@richtabor @annezazu done
<img width="355" alt="image" src="https://github.com/WordPress/gutenberg/assets/1157901/7fb6e37e-7c12-4374-ad5b-8c4b0b21ebdb">

jameskoster reviewed (approved):
Thank you, this looks good to go pending conflict resolution.

jameskoster commented:
Oh, I see that https://github.com/WordPress/gutenberg/pull/52497 got merged, and that's likely the source of the conflict 😅

I don't have a strong feeling either way on this one. How about you @richtabor?

vcanales commented:
This is redundant now since #52497 goes more in-depth. I suggest making further changes to the copy on another PR.
